crates/amaru/tests/evaluate_ledger_states.rs

```diff
@@ -0,0 +1,223 @@
+}
+
+fn get_test_context() -> Result<TestContext, Box<dyn std::error::Error>> {
+    let working_dir = env::current_dir()?;
+    Ok(TestContext {
+        ledger_dir: working_dir.join(PathBuf::from("../../ledger.db")),
+        pparams_dir: working_dir.join(PathBuf::from("../../cardano-blueprint/src/ledger/conformance-test-vectors/eras/conway/impl/dump/pparams-by-hash/")),
+    })
+}
+
+static TEST_MUTEX: Lazy<Mutex<()>> = Lazy::new(Mutex::default);
+
+include!("generated_ledger_conformance_test_cases.incl");
+
+fn evaluate_vector(snapshot: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let _shared = TEST_MUTEX.lock()?;
+    let network = NetworkName::Testnet(1);
+    let era_history = network.into();
+    match get_test_context() {
+        Ok(tc) => {
+            let vector_file = fs::read(snapshot)?;
+            let record: NESVector = minicbor::decode(&vector_file)?;
+            let () = import_vector(record, &tc.ledger_dir, era_history, &tc.pparams_dir)?;
+            Ok(())
+        }
+        Err(e) => {
+            tracing::warn!(
+                "skipping vector evaluation because some env vars are missing: {}",
+                e
+            );
+            Ok(())
+        }
+    }
+}
+
+fn peek_epoch(nes: &AnyCbor) -> Result<u64, Box<dyn std::error::Error>> {
+    let mut d = cbor::decode::Decoder::new(nes.raw_bytes());
+    let _ = d.array()?;
+    Ok(d.u64()?)
+}
+
+#[derive(cbor::Decode)]
+#[allow(dead_code)]
+struct NESVector {
+    #[n(0)]
+    initial_nes: AnyCbor,
+    #[n(1)]
+    final_nes: AnyCbor,
+    #[n(2)]
+    transactions: Vec<AnyCbor>,
+    #[n(3)]
+    test_state: String,
+}
+
+fn import_vector(
+    record: NESVector,
+    ledger_dir: &PathBuf,
+    era_history: &EraHistory,
+    pparams_dir: &PathBuf,
+) -> Result<(), Box<dyn std::error::Error>> {
+    if record.initial_nes[0] == 0x80 {
+        return Ok(());
+    }
+    let epoch = peek_epoch(&record.initial_nes)?;
+    let point = amaru_kernel::Point::try_from(
+        format!(
+            "{}.0000000000000000000000000000000000000000000000000000000000000000",
+            epoch * 86400
+        )
+        .as_str(),
+    )?;
+
+    fs::create_dir_all(ledger_dir)?;
+    let db = RocksDB::empty(ledger_dir)?;
+
+    let network = NetworkName::Testnet(1);
+    let epoch = crate::amaru_ledger::bootstrap::import_initial_snapshot(
+        &db,
+        &record.initial_nes,
+        &point,
+        network,
+        amaru_progress_bar::no_progress_bar,
+        Some(pparams_dir),
+        false,
+    )?;
+
+    let protocol_parameters = db.protocol_parameters()?;
+
+    let mut governance_activity = db.governance_activity()?;
+
+    let transaction = db.create_transaction();
+    transaction.save(
+        era_history,
+        &protocol_parameters,
+        &mut governance_activity,
+        &point,
+        None,
+        Default::default(),
+        Default::default(),
+        iter::empty(),
+    )?;
+    transaction.commit()?;
```

Comment:
What is happening here? I can only assume this is meant to set the _point_ as the tip of the ledger db, though that is already done by the `import_initial_snapshot` function. So that is probably redundant/pointless here.

Unless I am missing something, you can likely safely remove.